File: npython/hackerank_site/sort_string.py
#!/usr/bin/env python

"""
Author: Abhay Shanker Pathak
Date: 2020-03-20
Description: sort string
Email: abhaysp9955@gmail.com
Github: https://github.com/CoolAbhayS
Last Edited: 2020-03-20
"""

# non - pythonic way:

# declaring variabales: <<<
str1 = input('Enter the string: ')
# >>>

# converting str into list
# it's needed to create list as str doesn't supports item assignment
# lst = list(str1)

# sorting string: <<<
# for i in range(len(str1) - 1):
#     for j in range(i + 1, len(str1)):
#         if str1[i] > str1[j]:
#             lst[i], lst[j] = lst[j], lst[i]
# >>>

# print <<<
# print(f"Sorted string is {str1}")
# print('Sorted string is', ''.join(lst))
# >>>

# some pythonic approach <<<
# str1 decleration is at top
print('Sorted string is: ', ''.join(sorted(str1)))
# <<<

File: nc/prac1/pytoc/sort_string.c
/** Name -> Abhay Shanker Pathak
Date -> 2020-03-25
Description -> Sort string(sorting in ascending order)
Email -> abhaysp9955@gmail.com
Github -> https://github.com/CoolAbhayS
Last Edit -> 2020-03-25 */

#include<stdio.h>
#include<stdlib.h>
#include<string.h>

// main function <<<
int main(int argc, char *argv[]){
	//
	// variables <<<
	char *string, temp;
	int i, j;
	// >>>

	string = malloc(100 * sizeof(char));
	printf("\nEnter the string: ");
	scanf("%[^\n]%*c", string);
	/* scanf("%s", string); */ // this takes till space only

	/* printf("%c %c %d", *string, *(string + 1), *(string) == *(string + 1)); */

	for (i = 0; i < strlen(string) - 1; i++) {
		for (j = i + 1 ; j < strlen(string); j++) {
			/* if (*(string + j) == '\0') { */
			/* 	continue; */
			/* } */
			if (*(string + i) > *(string + j)) {
				temp = *(string + i);
				*(string + i) = *(string + j);
				*(string + j) = temp;
			}
		}
	}

	/* printf("%s", string); */
	for (i =0; i <= strlen(string); i++) {
		printf("%c", *(string + i));
	}

	return 0;
}
// >>>
